### PR TITLE
[NO_TICKET] Add Runtime.{Space, Store} type aliases

### PR DESCRIPTION
**Pull request by ArturGajowy**
## Overview
By capturing the set of parameters used in Runtime for
 - [C, P, A, R, K] (ISpace / RSpace / ReplayRSpace)
 - [C, P, A, K] (IStore / Context)
the aliases make code easier to read, write, and work with in general.

### JIRA issue:
n/a

### Checklist
- [x] This PR contains no more than 200 lines of code, excluding test code.
- [x] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [x] You have someone in mind to assign for review. Make this assignment after submitting the PR.

### CODE REVIEW

File: rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala [hunk 1]
```diff
@@ -20,23 +18,14 @@ import monix.eval.Task
...
-    var errorLog: ErrorLog,
-    val context: Context[Channel, BindPattern, ListChannelWithRandom, TaggedContinuation]) {
+class Runtime private (val reducer: Reduce[Task],
+                       val replayReducer: Reduce[Task],
+                       val space: Space[ISpace],
+                       val replaySpace: Space[ReplayRSpace],
+                       val costAccountingPure: CostAccountingAlg[Task],
+                       val costAccountingReplay: CostAccountingAlg[Task],
+                       var errorLog: ErrorLog,
+                       val context: Store[Context]) {
```
Comment (goral09) [inline rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala]: IDK if renaming `Context` to `Store` is a good change.

Comment (ArturGajowy) [reply rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala]: We can change the name. The current names were supposed to convey the meaning as follows:

`Repository.Space` - the set of parameters that Runtime uses for ISpace, RSpace, ReplayRSpace, and anything else that needs that exact quintuple of type parameters

`Repository.Store` - the set of parameters that Runtime uses for ISpace, and anything else that needs that exact quadruple of type parameters. E.g. Context.

I was thinking of using `CPARK` and `CPAK`, but dismissed those as cryptic.

Maybe there's a naming convention for type aliases like these?
`type FixParams[A[_]] = A[Int]` 
`type FixParams2[A[_, _]] = A[Int, String]`

Comment (ArturGajowy) [reply rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala]: Or, we could rename them to MyFoo, which would then excuse writing `MySpace[ISpace]` :stuck_out_tongue: (remember MySpace?)

Comment (ArturGajowy) [reply rholang/src/main/scala/coop/rchain/rholang/interpreter/Runtime.scala]: One more variant:
`StoreParams[IStore]`
`StoreParams[Context]`
`SpaceParams[Rspace]`

**Comment by ArturGajowy** [comment]
@itegulov @odeac @guardbotmk3 I'd like to merge this tomorrow morning, please provide any comments you might have till that time

**Pull request merged**